src/between.c

```diff
@@ -1,8 +1,24 @@
 #include "data.table.h"
 
+#define NA_INTEGER64 LLONG_MIN
+#define MAX_INTEGER64 LLONG_MAX
+
+bool isReallyRealC(SEXP x) {
```

Comment:
I wrote new `isReallyRealC` because I don't need allocating SEXP and I couldn't rewrite existing one because it returns location of first really real value, which is not what I need.
Ideally we could rename existing `isReallyReal` to `whichFirstReallyReal` and new `isReallyRealC` to `isReallyReal`. It will touch few other files so might be better to do it outside of this PR.